fenics_helpers/timestepping.py

```diff
@@ -42,7 +42,32 @@ def __del__(self):
         if self._show_bar:
             self._pbar.close()
 
+class CheckPoints:
+    def __init__(self, points, t_start, t_end, dt_min=1e-6):
+        self.points = np.array(points)
+        self.post_process_t0 = False
+        if self.points.size != 0:
```

Comment:
We could return early with
~~~
if not self.points:
    return
~~~
to flatten the method.